server/src/main/java/io/crate/expression/scalar/SubscriptFunction.java

```diff
@@ -169,7 +170,14 @@ public Object evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input[] a
         if (element == null || index == null) {
             return null;
         }
-        return lookup.apply(element, index);
+        try {
+            return lookup.apply(element, index);
+        } catch (ObjectKeyUnknownException e) {
```

Comment:
Could we adapt the lookup logic to not raise an exception if `errorOnUnknownObjectKey` is set to false? Given that this is run per row, performance here is important and creating stack traces for the exception is relatively expensive.